### PR TITLE
[bazel] Use GDB from Bazel instead of /tools

### PR DESCRIPTION
**Pull request by dmcardle**
Fixes https://github.com/lowRISC/opentitan/issues/15935

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -60,14 +60,15 @@ load("//third_party/cargo_raze:repos.bzl", "raze_repos")
...
-# load and instantiate `rules_foreign_cc_dependencies`.
-#load("@cargo_raze//:transitive_deps.bzl", "cargo_raze_transitive_deps")
-#cargo_raze_transitive_deps()
-load("@rules_foreign_cc//foreign_cc:repositories.bzl", "rules_foreign_cc_dependencies")
-rules_foreign_cc_dependencies()
+# The raze instructions would have us call `cargo_raze_transitive_deps`, but
+# that wants to re-instantiate rules_rust and mess up our rust configuration.
+# The single other action that transitive_deps would perform is to load and
+# instantiate `rules_foreign_cc_dependencies`, but this has already been done
+# above, so we can do nothing here.
```
Comment (drewmacrae) [inline WORKSPACE:67]: This feels like it should be split into a separate pr... but now I see this is a draft

Comment (dmcardle) [reply WORKSPACE:67]: Yep, this is a "dependent PR" — this first 3 commits will disappear when we merge #15853.

Comment (dmcardle) [reply WORKSPACE]: Parent is merged. Ready for review!

**Review by drewmacrae — APPROVED** [review]

**Review by alphan — APPROVED** [review]
